### PR TITLE
Add showvars plugin

### PR DESCRIPTION
**Pull request by jcpunk**
This plugin creates a quick cli interface to see what DNF vars are set for your dnf runtime. Folks using dnf vars (under /etc/dnf/vars) such as myself would benefit from a quick way of seeing what is set.

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -24,8 +24,10 @@ This documents extras plugins of DNF:
 .. toctree::
    :maxdepth: 1
 
+   kickstart
```
Comment (Conan-Kudo) [inline doc/index.rst]: Eh, why is `kickstart` added here?

File: doc/showvars.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
+..
+  Copyright (C) 2018  Red Hat, Inc.
```
Comment (Conan-Kudo) [inline doc/showvars.rst]: Do you work for Red Hat? If not, it's probably better to put your own copyright here...

File: plugins/showvars.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Show any DNF variables
+#
+# Copyright (C) 2018 Red Hat, Inc.
```
Comment (Conan-Kudo) [inline plugins/showvars.py]: Same here...

File: doc/index.rst (hunk 1 above)
Comment (jcpunk) [reply doc/index.rst]: Whoops, that slipped in....

I had a separate branch I was using to play with the doc stuff and noticed that kickstart wasn't listed in the documentation TOC.

Should kickstart be omitted from the doc TOC?

File: doc/showvars.rst (hunk 2 above)
Comment (jcpunk) [reply doc/showvars.rst]: Good call, I didn't look at the headers I copied really.  I'll get that fixed.

File: doc/index.rst (hunk 1 above)
Comment (Conan-Kudo) [reply doc/index.rst]: No, it should be there, just in added in a different PR. :)

**Comment by jcpunk** [comment]
Should be all fixed up now with a separate PR for the kickstart doc.

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot r+

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot retry

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot retry

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot retry

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot retry

**Comment by m-blaha** [comment]
There are rpm build errors on dnf-plugins-extras:
error: File not found: /builddir/build/BUILDROOT/dnf-plugins-extras-4.0.4-28g35da3b4.fc29.x86_64/etc/dnf/plugins/showvars.conf
error: File not found: /builddir/build/BUILDROOT/dnf-plugins-extras-4.0.4-28g35da3b4.fc29.x86_64/usr/share/man/man8/dnf.plugin.showvars.*
    File not found: /builddir/build/BUILDROOT/dnf-plugins-extras-4.0.4-28g35da3b4.fc29.x86_64/etc/dnf/plugins/showvars.conf
    File not found: /builddir/build/BUILDROOT/dnf-plugins-extras-4.0.4-28g35da3b4.fc29.x86_64/usr/share/man/man8/dnf.plugin.showvars.*

**Comment by jcpunk** [comment]
Just finally got back to this.  The rpmbuild bits should be fixed now.

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot retry

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot r+

**Comment by jcpunk** [comment]
I don't seem to be able to see why the test there failed....

**Comment by Conan-Kudo** [comment]
@jcpunk I don't know either, but the test system is broken anyway. LGTM, so merging.

**Pull request merged**